### PR TITLE
Adding linux VM staging tests to .ci.yaml

### PR DESCRIPTION
**Pull request by yusuf-goog**
Initial test with 2 tasks was successful, moving the
other tests in as a precursor to migration to
Ubuntu bots.

Bug: 102406

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by yusuf-goog** [comment]
@keyonghan the 2 tests previously checked in came in as successfully, and the dashboard hides them by default.

One uncovered the goldens mismatch (as expected for some tests), which was hidden because of other failure.

Moving all the linux VM ones to have them run as staging jobs.

These are VM tests and so not under the luci scheduler, which @CaseyHillers said would not badly impact `refresh-chrome-bot-status`

File: .ci.yaml [hunk 1]
```diff
@@ -2462,6 +2622,989 @@ targets:
       - bin/
       - .ci.yaml
 
+  - name: Staging_build_linux framework_tests_widgets
```
Comment (CaseyHillers) [inline .ci.yaml:2625]: nit: if we're going to have these long term, maybe the bottom of the ci.yaml should be the staging builders

**Review by CaseyHillers — APPROVED** [review]

File: .ci.yaml (hunk 1 above)
Comment (yusuf-goog) [reply .ci.yaml:2625]: Agreed. I'll move them.

**Review by keyonghan — DISMISSED** [review]
This will cause lots of flake noises affecting different team's triage process. We need to hold on this change before excluding these staging builders from the flake bot.
https://github.com/flutter/flutter/issues/103081#issuecomment-1117932570

**Comment by yusuf-goog** [comment]
This change now has ignore_flakiness and presubmit tags.

File: .ci.yaml [hunk 2]
```diff
@@ -4139,3 +4097,1247 @@ targets:
     properties:
       tags: >
           ["framework", "hostonly", "shard"]
+
+# Staging builds are specially constructed tasks.
+#
+# Names should begin with `Staging_build_` prefix.
+# Task should include presubmit: false
+# Task should inherit ignore_flakiness: true
```
Comment (keyonghan) [inline .ci.yaml:4105]: https://github.com/flutter/cocoon/pull/1828 is relying on target level property. So the platform property defined in this PR will not work.
We need to refactor the logic in https://github.com/flutter/cocoon/pull/1828 to incorporate https://github.com/flutter/cocoon/pull/1837.

**Review by keyonghan — CHANGES_REQUESTED** [review]

File: .ci.yaml (hunk 2 above)
Comment (yusuf-goog) [reply .ci.yaml:4105]: Added as target level property.

**Review by keyonghan — APPROVED** [review]
LGTM.

**Pull request merged**